plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/AmazonS3Fixture.java

```diff
@@ -26,71 +34,117 @@
 import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
 import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.rest.RestUtils;
-import org.elasticsearch.test.fixture.AbstractHttpFixture;
 
 import java.io.BufferedInputStream;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.nio.file.Files;
 import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
 
+import static com.carrotsearch.randomizedtesting.generators.RandomStrings.randomAsciiAlphanumOfLength;
+import static com.carrotsearch.randomizedtesting.generators.RandomStrings.randomAsciiAlphanumOfLengthBetween;
 import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
 
 /**
  * {@link AmazonS3Fixture} emulates an AWS S3 service
  * .
  * he implementation is based on official documentation available at https://docs.aws.amazon.com/AmazonS3/latest/API/.
  */
 public class AmazonS3Fixture extends AbstractHttpFixture {
+    private static final String AUTH = "AUTH";
+    private static final String NON_AUTH = "NON_AUTH";
+
+    private static final String EC2_PROFILE = "ec2Profile";
+
+    private final Properties properties;
+    private final Random random;
 
     /** List of the buckets stored on this test server **/
     private final Map<String, Bucket> buckets = ConcurrentCollections.newConcurrentMap();
 
     /** Request handlers for the requests made by the S3 client **/
     private final PathTrie<RequestHandler> handlers;
-    private final String permanentBucketName;
-    private final String temporaryBucketName;
 
     /**
      * Creates a {@link AmazonS3Fixture}
      */
-    private AmazonS3Fixture(final String workingDir, final String permanentBucketName, final String temporaryBucketName) {
+    private AmazonS3Fixture(final String workingDir, Properties properties) {
         super(workingDir);
-        this.permanentBucketName = permanentBucketName;
-        this.temporaryBucketName = temporaryBucketName;
+        this.properties = properties;
+        this.random = new Random(Long.parseUnsignedLong(requireNonNull(properties.getProperty("tests.seed")), 16));
+
+        new Bucket("s3Fixture.permanent", false, null);
+        new Bucket("s3Fixture.temporary", true, null);
+        final Bucket ec2Bucket = new Bucket("s3Fixture.ec2", false, random);
+
+        this.handlers = defaultHandlers(buckets, ec2Bucket.name);
```

Comment:
I was kind of thinking about creating the access key and token here and then passing them to `Bucket()` and `defaultHandlers()`, avoiding the `null-or-Random` parameter to `Bucket`, but that's ugly for other reasons. How about passing the bucket into `defaultHandlers` rather than passing its name and then looking it up again later?

Reply:
I thought we agreed to keep original way of auth check due to that edge case - I do see a profit of turn multipart delete into non-auth and do auth check there on demand